ixmp/reporting/__init__.py

```diff
@@ -548,3 +551,63 @@ def _config_args(path, keys, sections={}):
                   'have no effect').format(extra_sections))
 
     return result
+
+
+def _handle_computation_error(reporter, exc):
+    """Helper to print intelligible exception information for Reporter.get().
+
+    In order to aid in debugging, this helper:
+    - Omits the parts of the stack trace that are internal to dask, and
+    - Gives the key in the Reporter.graph and the computation task that
+      caused the exception.
+    """
+    from traceback import (
+        TracebackException,
+        format_exception_only,
+        format_list,
+    )
+    import sys
+
+    # Two copies of the stack: one with information on local variables (for
+    # retrieving the key and task); one without (for printing the traceback)
+    tb1 = TracebackException.from_exception(exc, capture_locals=True)
+    tb2 = TracebackException.from_exception(exc)
+
+    info = None  # Information about the call that triggered the exception
+    frames = []  # Frames for an abbreviated stack trace, from tb2
+    dask_internal = True  # Flag if the frame is internal to dask
+
+    # Iterate over frames from the base of the stack
+    for frame, summary in zip(tb1.stack, tb2.stack):
+        if frame.name == 'execute_task':
+            # Dask internal call to execute a task
+
+            # Retrieve information about the key/task that triggered the
+            # exception
+            info = dict(key=frame.locals['key'],
+                        task=frame.locals['task'])
+
+            # Remaining frames are related to the exception
+            dask_internal = False
+
+        if not dask_internal:
+            # Store the frame for printing the traceback
+            frames.append(summary)
+
+    # Assemble the exception printout
+
+    # Traceback; omitting a few dask internal calls below execute_task
+    lines = ['Traceback (most recent call last):\n'] + format_list(frames[3:])
+
+    # Type and message of the original exception
+    lines.extend(format_exception_only(exc.__class__, exc))
+
+    # Reporter information for debugging
+    lines.append("\nOccurred when computing {key!r} using:\n{task}\n\n"
```

Comment:
@OFR-IIASA as requested, now when exceptions get thrown the specific operation that generated them is printed out for easier debugging.